### PR TITLE
Add custom config and users tests

### PR DESCRIPTION
**Pull request by egegunes**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| semi-sync | passed |
| monitoring | passed |
| sidecars | failed |
| config | failed |
| users | passed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/27/commits/b62b477b5b91ffa0621a3b746fe20927b66df91a
image: `perconalab/percona-server-mysql-operator:PR-27-b62b477`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| semi-sync | passed |
| monitoring | passed |
| sidecars | failed |
| config | failed |
| users | passed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/27/commits/2ef5d99876afc0fa5cdf9fe208b4679cdab81d05
image: `perconalab/percona-server-mysql-operator:PR-27-2ef5d99`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| semi-sync | passed |
| monitoring | passed |
| sidecars | passed |
| config | failed |
| users | passed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/27/commits/2270311a6993d1d8d268dcd9fb263374269da33e
image: `perconalab/percona-server-mysql-operator:PR-27-2270311`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| semi-sync | passed |
| monitoring | passed |
| sidecars | passed |
| config | failed |
| users | passed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/27/commits/95c8aa12ceeb5472e30c467c9f374572c759f6c2
image: `perconalab/percona-server-mysql-operator:PR-27-95c8aa1`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| semi-sync | passed |
| monitoring | passed |
| sidecars | passed |
| config | passed |
| users | passed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/27/commits/96a878f74bbb74f99c56a8608c4658ec7cda9c2c
image: `perconalab/percona-server-mysql-operator:PR-27-96a878f`

**Review by tplavcic — DISMISSED** [review]

**Review by cap1984 — APPROVED** [review]
Just make linter happy

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| semi-sync | passed |
| monitoring | passed |
| sidecars | passed |
| config | passed |
| users | passed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/27/commits/863818cb614fb24cec99171c3f2f580186ca838d
image: `perconalab/percona-server-mysql-operator:PR-27-863818c`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| semi-sync | passed |
| monitoring | passed |
| sidecars | passed |
| config | passed |
| users | passed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/27/commits/ccfd2d24ab70e7e17b0997c9862bf1e886f2ecdb
image: `perconalab/percona-server-mysql-operator:PR-27-ccfd2d2`

**Review by cap1984 — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| semi-sync | passed |
| monitoring | passed |
| sidecars | passed |
| config | passed |
| users | passed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/27/commits/e09142efcabd5d5f82e9d2e1848af59f105a907e
image: `perconalab/percona-server-mysql-operator:PR-27-e09142e`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| semi-sync | passed |
| monitoring | passed |
| sidecars | passed |
| config | passed |
| users | passed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/27/commits/49be8ee18578c566ff426ed600b31e67b530104a
image: `perconalab/percona-server-mysql-operator:PR-27-49be8ee`

**Review by tplavcic — APPROVED** [review]
Maybe fix some of the shellcheck comments if appropriate, otherwise ok.

File: e2e-tests/functions [hunk 1]
```diff
@@ -23,19 +23,35 @@ deploy_client() {
...
+		| yq eval "$(printf '.spec.pmm.image="%s"' "${IMAGE_MYSQL}")" -
 }
 
 run_mysql() {
 	local command="$1"
 	local uri="$2"
+	local pod="$3"
 
-	kubectl -n "${NAMESPACE}" exec mysql-client -- \
+	kubectl -n "${NAMESPACE}" exec "${pod:-mysql-client}" -- \
```
Comment (cap1984) [inline e2e-tests/functions:54]: just for the sake of uniformity it's better to move variable defaults to variable declaration

File: e2e-tests/functions [hunk 2]
```diff
@@ -118,3 +134,13 @@ EOF
 	run_curl -XPOST -d "'$(echo ${payload} | sed 's/\n//g')'" "https://${user_pass}@${endpoint}/v0/qan/GetReport" \
 		| jq '.rows[].fingerprint'
 }
+
+get_mysql_pods() {
+	kubectl get pod -n "${NAMESPACE}" --no-headers --selector=app.kubernetes.io/component=mysql | awk '{print $1}'
```
Comment (cap1984) [inline e2e-tests/functions:139]:
```suggestion
	kubectl get pod -n "${NAMESPACE}" --no-headers --selector=app.kubernetes.io/component=mysql -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}'
```

**Review by cap1984 — APPROVED** [review]
LGTM in general
A few suggestions on bash

**Comment by egegunes** [comment]
I'm merging this. I'll apply your suggestions and fix all shellcheck comments on a separate PR @tplavcic @cap1984

**Pull request merged**